clientapi/auth/storage/accounts/postgres/storage.go

```
@@ -164,7 +164,7 @@ func (d *Database) createAccount(
 	}
 	if err := d.profiles.insertProfile(ctx, txn, localpart); err != nil {
 		if internal.IsUniqueConstraintViolationErr(err) {
-			return nil, nil
+			return nil, internal.ErrUserExists
```

You need to update `sqlite` as well please to return this sentinel error.